keps/sig-node/4216-image-pull-per-runtime-class/README.md

```diff
@@ -0,0 +1,1338 @@
+
+This problem can be fixed if containerd, CRI and kubelet refer to images as a tuple of (imageName/ID, runtimeClass) instead of the imageName/ID. With this, we can specify the runtimeClass a particular image is being run for and pull the appropriate manifest from the image index. This document details the changes needed to support pulling images per runtime class.
+Adding this support would enable us to have different versions of the same image on a given node and could be really beneficial in virtualization based runtimes. In windows, differnt versions of a multiarch image could be pulled based on hyperV and process isolation. It will also be very helpful to support future projects to support snapshotter per runtimeclass. 
+
+This KEP attempts to goes over the details of the most significant changes needed in containerd, CRI, kubelet and tools like ctr and crictl to support this feature.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Currently, the windows platform matcher looks for an exact OS version match between the host and guest while pulling images. This behavior is not accurate for hyperV isolation as the UVM/sandbox OS version could be different from the host OS version.  
+
+For example, a user might want to pull an image like python (a manifest list) on a WS2022 host with an intent of running it in on a WS2019 hyperV isolated UVM. But unfortunately, the current windows platform matcher behavior only forces the WS2022 version of the python image to be pulled (due to the exact version match between host and guest). If a manifest list does not have a version that matches the host OS version, then the image pull fails.
+Extending image pulls to specify an optional runtime class would greatly help to extend windows platform matcher behavior depending on whether hyperV or process isolation is being used. Additionally, having support to pull images per runtime class might also help projects like supporting different snapshotters with different runtimes as well.
```

Comment:
This section is mostly a duplicate of above, having it here is probably enough as the summary gives you the rational and this give more indepth details.